File: 7.reverse-integer/solution.java
class Solution {
    public int reverse(int x) {
        int sign = (int) Math.signum(x);
        x = Math.abs(x);
        long answer = 0;
        while (x > 0)
        {
        	answer = answer * 10 + x % 10;
        	x /= 10;
        }
        answer *= sign;
        return answer > Integer.MAX_VALUE || answer < Integer.MIN_VALUE ? 0 : (int) answer;
    }
}

File: 7.reverse-integer/solution.py
class Solution:
    def reverse(self, x: int) -> int:
        result = f(x) if x > 0 else -f(-x)
        return result if result < 2147483648 and result >= -2147483648 else 0

def f(x: int) -> int:
    re = 0
    while x > 0:
        re = re * 10 + x % 10
        x //= 10
    return re
